### PR TITLE
Added docker jdk21 build configuration

### LINKED ISSUE #7899: Add JDK 21 docker image
JDK21 potentially brings performance improvements without changing code, and potentially better garbage collection logic. we should create a JDK-21 image so it can be tested out.

### PR DESCRIPTION
**Pull request by rolfyone**
I'm not sure if this publishes, will have to check when we get to develop maybe...

I started up a minimal node and was able to produce blocks.

fixes #7899

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by tbenr — APPROVED** [review]
LGTM. @benjaminion effects are already playing out

File: docker/jdk17/Dockerfile [hunk 1]
```diff
@@ -46,8 +46,8 @@ ARG VCS_REF
 ARG VERSION
 LABEL org.label-schema.build-date=$BUILD_DATE \
       org.label-schema.name="Teku" \
-      org.label-schema.description="Ethereum 2.0 Beacon Chain Client" \
-      org.label-schema.url="https://consensys.net/knowledge-base/ethereum-2/teku/" \
+      org.label-schema.description="Ethereum Consensus Client" \
+      org.label-schema.url="https://docs.teku.consensys.io/" \
```
Comment (StefanBratanov) [inline docker/jdk17/Dockerfile:50]: I think this is better link marketing wise: https://consensys.io/teku

**Review by StefanBratanov — COMMENTED** [review]
just a nit

File: docker/jdk17/Dockerfile (hunk 1 above)
Comment (rolfyone) [reply docker/jdk17/Dockerfile:50]: it had all the wrong terminology but the old link was just plain wrong... at least the doc link is maintained... its a good start and we can revisit.

**Pull request merged**